File: awesome_map_mobile/lib/models/problem/problem.dart
import 'dart:io';

import 'package:awesome_map_mobile/models/base/category.dart';
import 'package:awesome_map_mobile/models/comment/comment.dart';
import 'package:awesome_map_mobile/models/files/fileBody.dart';
import 'package:awesome_map_mobile/models/files/serverFile.dart';
import 'package:awesome_map_mobile/models/problem/problemStatus.dart';
import 'package:json_annotation/json_annotation.dart';
part 'problem.g.dart';

@JsonSerializable()
class Problem {
  @JsonKey(required: true)
  String id;
  @JsonKey(required: true)
  double latitude;
  @JsonKey(required: true)
  double longitude;
  @JsonKey(required: true)
  String title;
  @JsonKey(required: true)
  List<Category> problemTypes;
  @JsonKey(required: true)
  int status;
  @JsonKey(required: true)
  DateTime createDate;
  DateTime updateDate;
  List<Comment> comments;
  int commentsLength;

  String description;
  String ownerId;
  List<ServerFile> files;
  int subscribersCount;
  Problem(
      this.id,
      this.latitude,
      this.longitude,
      this.title,
      this.status,
      this.createDate,
      this.updateDate,
      this.description,
      this.ownerId,
      this.problemTypes,
      this.files,
      this.comments,
      this.commentsLength,
      this.subscribersCount);

  factory Problem.empty() {
    return Problem(
        "00000000-0000-0000-0000-000000000000",
        0,
        0,
        "",
        0,
        DateTime.now(),
        null,
        "",
        "00000000-0000-0000-0000-000000000000",
        new List<Category>(),
        new List<ServerFile>(),
        new List<Comment>(),
        0,
        0);
  }

  factory Problem.fromJson(Map<String, dynamic> json) =>
      _$ProblemFromJson(json);

  Map<String, dynamic> toJson() => _$ProblemToJson(this);

  addCategory(Category item) {
    this.problemTypes.add(item);
  }

  removeCategory(String guid) {
    this.problemTypes =
        this.problemTypes.where((item) => item.id != guid).toList();
  }

  
}

File: awesome_map_server/DataBaseModels/Problem/Problem.cs
using DataBaseModels.Models.Base;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace DataBaseModels.Models {
    public class Problem : Ticket {
        public Guid Id { get; set; }
        public string Title { get; set; }
        public virtual List<ProblemTypeProblem> ProblemTypeProblems { get; set; }
        public string Description { get; set; }
        public string OwnerId { get; set; }
        public virtual ApplicationUser Owner { get; set; }
        public virtual List<ProblemUser> Subscribers { get; set; }
        public ProblemStatus Status { get; set; } = ProblemStatus.Open;
        public virtual List<ServerFile> Files { get; set; }
        public virtual List<Comment> Comments { get; set; }
        public DateTime CreateDate { get; set; }
        public DateTime? UpdateDate { get; set; }


        Problem() {
            ProblemTypeProblems = new List<ProblemTypeProblem>();
            Subscribers = new List<ProblemUser>();
            Comments = new List<Comment>();
        }

    }
}
